### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by guits**
- update doc
- clean ceph.conf.j2 template
- fix bz https://bugzilla.redhat.com/show_bug.cgi?id=1580408
- deprecate fqdn deployment

### CODE REVIEW

File: roles/ceph-defaults/tasks/facts.yml [hunk 1]
```diff
@@ -217,3 +217,22 @@
...
+      register: ceph_current_status
+
+    - name: set_fact ceph_current_status (convert to json)
+      set_fact:
+        ceph_current_status: "{{ ceph_current_status.stdout | from_json }}"
+
+    - name: set_fact rgw_servicemap
+      set_fact:
+        rgw_servicemap: "{% for key in ceph_current_status['servicemap']['services']['rgw']['daemons'].keys() %}{% if key == ansible_fqdn %}{{ key }}{% endif %}{% endfor %}"
+
```
Comment (leseb) [inline roles/ceph-defaults/tasks/facts.yml]: why this empty line?

File: roles/ceph-defaults/tasks/facts.yml [hunk 2]
```diff
@@ -217,3 +217,22 @@
...
+  when:
+    # no servicemap before luminous
+    - ceph_release_num[ceph_release] >= ceph_release_num['luminous']
+    - ansible_hostname != ansible_fqdn
+    - ceph_current_fsid.get('rc', 1) == 0
```
Comment (leseb) [inline roles/ceph-defaults/tasks/facts.yml]: shouldn't we move this on top instead?

File: roles/ceph-defaults/defaults/main.yml [hunk 3]
```diff
@@ -603,3 +603,9 @@ openstack_keys:
   - { name: client.cinder-backup, caps: { mon: "profile rbd", osd: "profile rbd pool={{ openstack_cinder_backup_pool.name }}"}, mode: "0600" }
   - { name: client.gnocchi, caps: { mon: "profile rbd", osd: "profile rbd pool={{ openstack_gnocchi_pool.name }}"}, mode: "0600", }
   - { name: client.openstack, caps: { mon: "profile rbd", osd: "profile rbd pool={{ openstack_glance_pool.name }}, profile rbd pool={{ openstack_nova_pool.name }}, profile rbd pool={{ openstack_cinder_pool.name }}, profile rbd pool={{ openstack_cinder_backup_pool.name }}"}, mode: "0600" }
+
```
Comment (leseb) [inline roles/ceph-defaults/defaults/main.yml]: please add extra empty line

**Comment by guits** [comment]
jenkins test pipeline

**Pull request merged**